aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -221,6 +223,37 @@
   @Deprecated
   public static final boolean CLIENT_ENABLE_ETAG_CHECK_DEFAULT = false;
 
+  /**
+   * If set, the S3 client provided by {@link LakeFormationAwsClientFactory}
+   * will use lake formation provided credentials.
+   */
+  public static final String S3_USE_LAKE_FORMATION_CREDENTIALS = "s3.use.lakeformation.credentials";
+
+  /**
+   * The table name used as part of lake formation credentials request.
+   */
+  public static final String LAKE_FORMATION_TABLE_NAME_KEY = "lakeformation.table.name";
```

Comment:
dot is used to represent levels in config names, dash is used to connect words, so in this case it's better to name it as something like `lakeformation.table-name`